### PR TITLE
Datasource: Fixes clicking outside of some query editors required 2 clicks

### PR DESCRIPTION
**Pull request by aocenas**
Slate handling of blur events is problematic and so for example when clicking into query editor in explore and then clicking on datasource select did not work on first click. This seems to be due to some issue in Slate editor that refocuses on rerender. This fixes this issue by removing the need to rerender. In addition this adds some comments and does some cleanup.

### CODE REVIEW

**Review by davkal — CHANGES_REQUESTED** [review]
Fixes focus issue 🎉 

Blocker: Explore must not run queries on blur. Queries can take 10s of secs, and execution should be an active step.

File: public/app/features/explore/QueryField.tsx [hunk 1]
```diff
@@ -15,18 +15,18 @@ import IndentationPlugin from './slate-plugins/indentation';
...
-
 export interface QueryFieldProps {
   additionalPlugins?: Plugin[];
   cleanText?: (text: string) => string;
   disabled?: boolean;
-  initialQuery: string | null;
+  // We have both value and local state. This is usually an antipattern but we need to keep local state
+  // for perf reasons and also have outside value in for example in Explore redux that is mutable from logs
+  // creating a two way binding.
+  value: string | null;
```
Comment (davkal) [inline public/app/features/explore/QueryField.tsx]: I'd like to avoid a confusion with Slate values.

**Review by davkal — APPROVED** [review]

**Comment by davkal** [comment]
Unblocking PR, run on blur was not a recent regression.

File: public/app/features/explore/QueryField.tsx (hunk 1 above)
Comment (aocenas) [reply public/app/features/explore/QueryField.tsx]: Changed to `query` mainly just wanted to remove the `initial` as it usually suggests that after initial render it should not affect the value which is not true here.

**Comment by aocenas** [comment]
> Blocker: Explore must not run queries on blur. Queries can take 10s of secs, and execution should be an active step.

Ah sorry I did not understood that from out call, though that was just a possible way how to fix it not a final desired behaviour, can change in separate PR for 6.5

**Pull request merged**